### DISCUSSION TITLE
d.ts files do not trigger type-check

### DISCUSSION THREAD

**Discussion by qqstt** [web-infra-dev/rsbuild#3006] [Q&A]
when i use @rsbuild/plugin-type-check and change d.ts, the plugin not work
`  
plugins: [
    pluginTypeCheck({
      forkTsCheckerOptions: {
        typescript: {
          mode: 'readonly',
          memoryLimit: 4096,
        },
      },
    }),
    pluginBabel({
      include: /\.(?:jsx|tsx)$/,
    }),
    pluginVue(),
    pluginVueJsx(),
    pluginSass(),
  ],
`

**Comment by chenjiahan** [comment]
Have you included these dts files in your tsconfig.json?

**Reply by qqstt** [reply]
yes,
```
  "include": [
    "src/**/*.ts",
    "src/**/*.d.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "../magicTypes/types/**/*",
    "types/**/*.d.ts"
  ],

```

**Reply by chenjiahan** [reply]
You can run "npx tsc" and see if tsc will check it.

**Reply by qqstt** [reply]
I delete all files only leave main.ts and test.d.ts, and run `npx tsc -watch`，I change test.d.ts and  it work;  but i run `rsbuild dev`  and  change test.d.ts it not work
